### DISCUSSION TITLE
ScrollContainer - option to change side of scrollbar

### DISCUSSION THREAD

**Discussion by FireCatMagic** [godotengine/godot-proposals#10598] [GUI]
The scrollbar for it is always on the right in vertical mode, and bottom in horizontal. It would be nice if it could be set to go on either the top/bottom for horizontal and right/left for vertical instead of being locked to one side.

**Comment by AThousandShips** [comment]
I can't really think of any ui I've seen where the scroll bar is on the left for a LTR UI, it will be on the left for RTL UI, this would be very unorthodox and I don't expect many people to ever use it

**Reply by FireCatMagic** [reply]
While usually it wouldn't go on the left, it would be useful for non traditional use cases, such as in my game which has an element with two scroll containers, one on the left and on the right. Users make levels in the game and the containers will fill up with all the stars in the level, and if the user adds a lot there will be enough to make the scrollbars pop up. It ends up with the scrollbars both being on the right, where it would look a lot cleaner if the left sides scrollbar was also on the left, so it'd be symmetrical

**Reply by AThousandShips** [reply]
All UI I've seen with two scroll areas next to each other has used scroll bars on the same side, including the Godot editor

What I'm saying is that this is *very* niche and adding a bunch of extra code to the engine for something *very few* will use is against our policy for features

**Reply by timothyqiu** [reply]
It may be just a matter of UI standard or historically used convention. For example,

- this [screenshot](https://en.wikipedia.org/wiki/NeXT#/media/File:NeXTSTEP_desktop.png) of NeXTSTEP desktop on Wikipedia is in English and the vertical scrollbars are all on the left side.
- this version of [Oracle Applications User Interface Standards](https://docs.oracle.com/cd/A60725_05/pdf/uistan.pdf), published in 1998, specifies that scroll bars should be "placed to the left of the block" (page 19). A relevant standard published more recently changed that to the right side.

I can understand how someone might put the scrollbar on the left side for visual style purposes, or simply want to mimic the behavior of legacy systems, e.g., for retro games.

**Reply by AThousandShips** [reply]
I'd say that *if* we do anything to support this it'd be most useful to configure `ScrollContainer` (and other `Control`s that has scroll in them like this) to use a custom scroll bar or scroll bars entirely, with a `set_custom_h_scroll(ScrollBar)` method and similar which replaces the internal scroll bar with an external one, then you can redesign or reposition it as you wish, I think adding one more layer of minor configuration support for `ScrollContainer` (but not things like `RichTextLabel`, `TextEdit`, etc., or to them *as well*, making the work even more extensive) is not realistic, and it would be more constructive to separate out the configuration entirely

**Reply by AThousandShips** [reply]
Opened an issue for this:
* https://github.com/godotengine/godot-proposals/issues/10620
